pkg/eksd/installer.go

```diff
@@ -41,22 +41,24 @@ func NewEksdInstaller(client EksdInstallerClient, reader Reader) *Installer {
 
 func (i *Installer) InstallEksdCRDs(ctx context.Context, clusterSpec *cluster.Spec, cluster *types.Cluster) error {
 	var eksdCRDs []byte
-	if err := i.retrier.Retry(
-		func() error {
-			var readerErr error
-			eksdCRDs, readerErr = i.reader.ReadFile(clusterSpec.VersionsBundle.EksD.Components)
-			return readerErr
-		},
-	); err != nil {
-		return fmt.Errorf("loading manifest for eksd components: %v", err)
-	}
+	for _, vb := range clusterSpec.Bundles.Spec.VersionsBundles {
+		if err := i.retrier.Retry(
+			func() error {
+				var readerErr error
+				eksdCRDs, readerErr = i.reader.ReadFile(vb.EksD.Components)
```

Comment:
`EksD.Components` are probably the same for all version bundles. I would suggest creating a set with the urls and avoid duplicating calls

Reply:
Very good point. Updated